pwd_guess.py

```diff
@@ -1711,8 +1744,10 @@ def do_calculate_probs_from_file(self):
                     for i in range(1, self.config.probability_striation + 1)]
         pwds = self.read_test_passwords()
         logging.info('Calculating test set probabilities')
-        #todo decide here which probabilitycalculator to use
-        return ManyToManyProbabilityCalculator(self).calc_probabilities(pwds)
+        if self.config.sequence_model == Sequence.MANY_TO_MANY:
```

Comment:
Any class that you change should also have added tests to account for the new code paths added by the change. So for example, the Guesser class should have a test added for when using the many to many model, it can still return the correct output for the conditional_probs_many, batch_prob, and super_node_recur functions. The ManyToManyTrainer should also have a test for its new next_train_set_as_np, and the ManyToManyProbabilityCalculator should have a test added. Finally, we should add an integration test like the other tests in TestMainConfigurations that test the new model.

Reply:
I added a few tests. Should I add tests for functions like conditional_probs_many, super_node_recur, probability_stream given that they were not tested before?